solver/src/main/java/org/chocosolver/solver/constraints/nary/alldifferent/conditions/PropCondAllDiffBase.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * This file is part of choco-solver, http://choco-solver.org/
+ *
+ * Copyright (c) 2022, IMT Atlantique. All rights reserved.
+ *
+ * Licensed under the BSD 4-clause license.
+ *
+ * See LICENSE file in the project root for full license information.
+ */
+package org.chocosolver.solver.constraints.nary.alldifferent.conditions;
+
+import org.chocosolver.solver.constraints.Propagator;
+import org.chocosolver.solver.constraints.PropagatorPriority;
+import org.chocosolver.solver.variables.IntVar;
+
+/**
+ * Base class for conditional AllDifferent propagators.
+ *
+ * @author Dimitri Justeau-Allaire.
+ */
+public abstract class PropCondAllDiffBase extends Propagator<IntVar> {
+
+    protected final Condition condition;
+
+    public PropCondAllDiffBase(IntVar[] variables, Condition condition, PropagatorPriority priority) {
+        super(variables, priority, false);
+        this.condition = condition;
+    }
+
+    /**
+     * @return The variables respecting the condition.
+     */
+    protected IntVar[] filterVariables() {
+        int nb = 0;
+        for (IntVar v : vars) {
```

Comment:
Why the condition is checked twice?
Wouldn't be simpler to use a stream of variables and filter them on the basis of the condition?

Reply:
I just repeated what is currently done Choco: https://github.com/chocoteam/choco-solver/blob/531ab5dde8e86348181e72af46bc14bd334906db/solver/src/main/java/org/chocosolver/solver/constraints/nary/alldifferent/conditions/PropCondAllDiff_AC.java#L53

But I agree that a stream + a filter would be better, I fix this right now.